src/build.d

```diff
@@ -801,6 +885,146 @@ void processEnvironment()
+    {
+        warnings = [
+            "-Wall",
+            "-Wextra",
+            "-Werror",
+            "-Wno-attributes",
+            "-Wno-char-subscripts",
+            "-Wno-deprecated",
+            "-Wno-empty-body",
+            "-Wno-format",
+            "-Wno-missing-braces",
+            "-Wno-missing-field-initializers",
+            "-Wno-overloaded-virtual",
+            "-Wno-parentheses",
+            "-Wno-reorder",
+            "-Wno-return-type",
+            "-Wno-sign-compare",
+            "-Wno-strict-aliasing",
+            "-Wno-switch",
+            "-Wno-type-limits",
+            "-Wno-unknown-pragmas",
+            "-Wno-unused-function",
+            "-Wno-unused-label",
+            "-Wno-unused-parameter",
+            "-Wno-unused-value",
+            "-Wno-unused-variable"
+        ];
+
+        if(cxxKind == "g++") {
+            warnings ~= [
+                "-Wno-logical-op",
+                "-Wno-narrowing",
+                "-Wno-unused-but-set-variable",
+                "-Wno-uninitialized",
+                "-Wno-class-memaccess",
+                "-Wno-implicit-fallthrough"
+            ];
+        }
+    }
+    else
+    {
+        warnings = [
+            "-Wno-deprecated",
+            "-Wstrict-aliasing",
+            "-Werror"
+        ];
+
+        if(cxxKind == "clang++")
+            warnings ~= "-Wno-logical-op-parentheses";
+    }
+
+    string hostDmdVernum;
+
+    if(env["HOST_DMD_KIND"].among("dmd", "ldc"))
+    {
+        import std.uni: isNumber;
+
+        auto json = run([ env["HOST_DMD_RUN"], "-Xf-", "-Xi=compilerInfo" ]).output;
+
+        // New JSON format
+        if (json.findSkip(`"__VERSION__" : `)) // dddd
+        {
+            hostDmdVernum = json.until!(c => !isNumber(c)).to!string;
+        }
+        // Old JSON format:
+        else if (json.findSkip(`"version" : "v2.`)) // "ddd"
+        {
+            hostDmdVernum = "2".chain(json.until!(c => !isNumber(c))).to!string;
+        }
+        else
+            assert(false, "Failed to detect HOST_DMD_VERSION from: " ~ json);
+    }
+    else // GDC does not support compilerInfo (yet)
+    {
+        enum redirect = Redirect.stdin | Redirect.stdout | Redirect.stderrToStdout;
+        auto pipes = pipeProcess([env["HOST_DMD_RUN"], "-o-", "-c", "-"], redirect);
+
+        pipes.stdin.writeln("pragma(msg, cast(int)__VERSION__);");
+        pipes.stdin.flush();
+        pipes.stdin.close();
+
+        // wait(pipes.pid); // gdmd exits != 0 without source files
+
+        hostDmdVernum = pipes.stdout.byLine.front.idup;
+    }
+
+    auto cxxFlags = warnings ~ [
+        "-g", "-g3", "-DDEBUG=1",
+        "-DUNITTEST",
+        "-fno-exceptions",
+        "-fno-rtti",
+        "-DMARS=1",
+        env["MODEL_FLAG"],
+        env["PIC_FLAG"],
+        "-DDMD_VERSION=" ~ hostDmdVernum,
+
+        // No explicit if since cxxKind will always be either g++ or clang++
+        cxxKind == "g++" ? "-std=gnu++98" : "-xc++"
```

Comment:
`cxxKind = "-std=gnu++98"`

Reply:
?

Reply:
I believe that the expression `"g++"` will always evaluate to `true` so this ternary expression will always return `"-std=gnu++98"`.

Reply:
No,e.g Linux_64_64 on the autotester run clang

Reply:
goodness I thought the `==` was an `=`.  ignore this comment

Reply:
Thankfully `const` prevents such debugging nightmares.